### ISSUE TITLE
wpgtk doesn't seem to work with terminal backgrounds on Wayland

### ISSUE THREAD

**Issue by harrisonthorne** [deviantfero/wpgtk#130]
It seems like wpgtk doesn't like to set the background of terminals on Wayland.

I'm on Arch Linux (kernel 4.20) and using Sway 1.0 beta, wpgtk version 5.8.0, and fish-shell 3.0.0 (by the way, what's the fish equivalent of `(wpg -t &)`? that might help)

When running `wpg -t` in a terminal (I've tested all sorts of terminal emulators), the text colors will change, but the background will not.

**Comment by deviantfero** [comment]
what terminal are you using? also, (wpg -t &) is deprecated, you should be using (cat ~/.config/wpg/sequences), this removes the overhead of starting up python and loads the colors faster.

**Comment by harrisonthorne** [comment]
I use alacritty. I've also tested the following:

Emulators whose backgrounds don't change:
* alacritty
* kitty
* gnome-terminal
* Konsole (no color change at all)

It seems like termite might be working, but I can't tell. I'm not really sure about termite.

**Comment by GotekSC** [comment]
This is also the case on X11 in my testing of
+ alacritty
+ kitty
+ termite

There's a special case with st which changes BG color on initial startup with xresources patch.

Not sure if I should open a new Issue for this.

**Comment by harrisonthorne** [comment]
Looks like the problem disappeared for me; I totally forgot that this thread exists (haha, my memory needs help) I think `cat ~/.config/wpg/sequences` might have solved the problem for me. I don't remember changing anything else, except possibly `ln -s ~/.Xresources ~/.Xdefaults`. 

Try both of those @GotekSC and see if it solves your issue. I'm going to close this issue since it's specific to Wayland. :) Unless @deviantfero wants to handle both display protocols under this issue

**Comment by harrisonthorne** [comment]
This issue has resurfaced again. Reopening.

**Comment by deviantfero** [comment]
Hi, sorry for taking so long, I've not been able to reproduce this problems in any of these terminals, does this problem only appear on `wayland`? can you confirm?

**Comment by deviantfero** [comment]
Have you tried these steps?

https://github.com/dylanaraps/pywal/wiki/Customization#terminal-emulators

remember that `wpgtk` uses `pywal` as a backend, so you should have those files readily available on each color-scheme change done by wpgtk too

**Comment by sseneca** [comment]
I'm getting the same issue. `wpgtk` works fine on Arch/X11 but not on Void/Wayland. The `pywal` instructions, `ln -s ~/.Xresources ~/.Xdefaults`, and `cat ~/.config/wpg/sequences` didn't fix the issue. I tested these terminals:

- `kitty`
- `alacritty`
- `xterm`
- `urxvt`
- `st`

Also, logging into the Void/Wayland machine via ssh from the Arch/X11 machine would set the (incorrect) colour scheme, as demonstrated [in this video](https://libre.video/videos/watch/067edc85-951a-47a9-a5e7-3797287d412a).

**Comment by sseneca** [comment]
Update: for me, the problem is solved.

I installed `xrdb` which I didn't have on the Void/Wayland machine at first. `xrdb -query` outputted nothing whereas the same command on Arch outputs a bunch of colour definitions.

I then ran `wal --theme solarized`, which worked as expected (that is, it set the terminal theme correctly). Now the `wpg` command (`wpg -s dark.png`) is working too. The `wpg` GUI would still set the wrong scheme though, so I manually replaced the `json` colour scheme files (in the `~/.config/wpg/schemes` directory) with those generated on my Arch/X machine as well. I'm not sure if that's necessary though, it probably needs more experimentation.
